label_studio/core/utils/io.py

```diff
@@ -208,3 +210,20 @@ def validate_upload_url(url, block_local_urls=True):
         if ipaddress.ip_address(ip) in ipaddress.ip_network(subnet):
             raise InvalidUploadUrlError
 
+def generate_key_if_missing(key):
+    value = env.str(key, "")
+
+    if value == "":
+        print(f'Warning: {key} not found in environment variables will generate a random key')
+        value = get_random_secret_key()
+        try:
+            env_file = os.path.join(base_dir, '..', '..', '.env')
+            with open(env_file, 'a') as f:
+                f.write(f'\n{key}={value}')
+        except Exception as e:
+            print(f'Warning: failed to write {key} to .env file: {e}, new key will be regenerated on every run')
```

Comment:
Maybe add more context on the impact about it being generated every run? Sessions will be reset at every restart, encrypted values (do we have that in LS?) will be undecryptable... Does this affect password hashes?

Reply:
https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-SECRET_KEY <- "Secret keys are not used for passwords of users and key rotation will not affect them."

adding more context here SGTM also